atomicapp/cli/main.py

```diff
@@ -100,43 +99,75 @@ def cli_stop(args):
 
     def __init__(self):
-        self.parser = ArgumentParser(
-            prog='atomicapp',
-            description=(
-                "This will install and run an Atomic App, "
-                "a containerized application conforming to the Nulecule Specification"),
-            formatter_class=RawDescriptionHelpFormatter)
+        self.parser = self.create_parser()
 
-    def set_arguments(self):
+    def create_parser(self):
 
-        base_parser = ArgumentParser(add_help=False)
+        # We will have a few parsers that we use. The toplevel parser
+        # will be the parser that ultimately gets called. It will consist
+        # of subparsers for each "action" and each of those subparsers will
+        # inherit from a parser for all global options.
 
-        base_parser.add_argument(
+        # === TOPLEVEL PARSER ===
+        # Create the toplevel parser. This is the one we will return
+        toplevel_parser = argparse.ArgumentParser(
+            prog='atomicapp',
+            formatter_class=argparse.RawDescriptionHelpFormatter,
+            add_help=False,
+            description=(
+                "This will install and run an Atomic App, "
+                "a containerized application conforming to the Nulecule Specification"))
+        # Add a help function to the toplevel parser but don't output
+        # help information for it. We need this because of the way we
+        # are stitching help output together from multiple parsers
+        toplevel_parser.add_argument(
+            "-h",
+            "--help"
+            "--version",
+            action='help',
+            help=argparse.SUPPRESS)
+        # Allow for subparsers of the toplevel_parser. Store the name
+        # in the "action" attribute
+        toplevel_subparsers = toplevel_parser.add_subparsers(dest="action")
+
+        # === GLOBAL OPTIONS PARSER ===
+        # Create the globals argument parser next. This will be a
+        # parent parser for the subparsers
+        globals_parser = argparse.ArgumentParser(add_help=False)
+        globals_parser.add_argument(
             "-V",
             "--version",
             action='version',
             version='atomicapp %s, Nulecule Specification %s' % (
                 __ATOMICAPPVERSION__, __NULECULESPECVERSION__),
             help="show the version and exit.")
-        # TODO refactor program name and version to some globals
-
-        base_parser.add_argument(
+        globals_parser.add_argument(
             "-v",
             "--verbose",
             dest="verbose",
             default=False,
             action="store_true",
             help="Verbose output mode.")
-
-        base_parser.add_argument(
+        globals_parser.add_argument(
             "-q",
             "--quiet",
             dest="quiet",
             default=False,
             action="store_true",
             help="Quiet output mode.")
-
-        base_parser.add_argument(
+        globals_parser.add_argument(
```

Comment:
Probably not, it's only really needed for one path (i.e. run would probably be the only way we ever use it).. the one thing I do like about having it as a global is that it will show in the help when someone types `atomcapp --help`.

I can move it to a non-global if you like.